botfront/imports/api/graphql/botResponses/schemas/botResponses.types.graphql

```diff
@@ -37,6 +37,7 @@ type Mutation {
         language: String!
         newPayload: Any
         key: String!
+        setType: String
```

Comment:
I could not find where the setType variable was used on the client side, from the botresponse container the updateResponse does not seems to use it
```
const updateResponse = (updatedResponse, callback = () => {}) => {
        updateBotResponse({
            variables: {
                projectId, _id: updatedResponse._id, response: clearTypenameField(updatedResponse),
            },
. 
.
.
```
Can you shine some light about how it's working ?

minor : I think setType can make you think it's a function, so maybe just type or responseType ?

Reply:
I renamed set type to newResponseType which should better communicate what it is.

so newResponseType is used by the upsertResponse mutation called in project.jsx line 169 to 189
if the response type has changed it adds the updateResponseType field to the graphQL variables
in the upsertResponse resolver, if the updateResponseType is set it will use the updateResponseType function from responses.js instead of the upsert response function

I could separate the response type update out into its own mutation/resolver if that would be better